supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActor.scala

```diff
@@ -90,6 +204,22 @@ class TesAsyncBackendJobExecutionActor(override val standardParams: StandardAsyn
     )
   }
 
+  override def scriptPreamble: String = {
+    val tesTaskPreamble: String = runtimeAttributes.localizedSasEnvVar match {
+      case Some(environmentVariableName) =>
+        val workflowName = workflowDescriptor.callable.name
+        val callInputFiles = jobDescriptor.fullyQualifiedInputs.safeMapValues {
+        _.collectAsSeq { case w: WomFile => w }
+        }
+        val taskInputs: List[Input] = TesTask.buildTaskInputs(callInputFiles, workflowName, mapCommandLineWomFile)
+        determineWSMSasEndpointFromInputs(taskInputs, getPath, jobLogger).map { endpoint =>
+        generateLocalizedSasScriptPreamble(environmentVariableName, endpoint)
```

Comment:
Hmm, should we continue attempting to run the task if we can't get the SAS URL and generate the preamble? I'm not sure which way would be better. If we do continue running it, it might be good to log a message containing the failure from `determineWSMSasEndpointFromInputs`

Reply:
Good point. I think that since the user specified a `runtimeAttribute` that Cromwell can't fulfill, the task should fail.

Better to fail fast in cromwell than wait for the task to fail due to lack of expected token.